logprep/framework/pipeline.py

```diff
@@ -158,24 +158,23 @@ def _create_metrics(self):
         )
 
     def _build_pipeline(self):
-        if self._logger.isEnabledFor(DEBUG):  # pragma: no cover
-            self._logger.debug(f"Building '{current_process().name}'")
-        self._pipeline = []
-        for entry in self._logprep_config.get("pipeline"):
-            processor_name = list(entry.keys())[0]
-            entry[processor_name]["metric_labels"] = self._metric_labels
-            processor = Factory.create(entry, self._logger)
-            self._pipeline.append(processor)
-            self.metrics.pipeline.append(processor.metrics)
-            if self._logger.isEnabledFor(DEBUG):  # pragma: no cover
-                self._logger.debug(f"Created '{processor}' processor ({current_process().name})")
-            self._pipeline[-1].setup()
-        if self._logger.isEnabledFor(DEBUG):  # pragma: no cover
-            self._logger.debug(f"Finished building pipeline ({current_process().name})")
+        self._logger.debug(f"Building '{current_process().name}'")
+        self._pipeline = tuple(
+            (self._create_processor(entry) for entry in self._logprep_config.get("pipeline"))
+        )
+        self._logger.debug(f"Finished building pipeline ({current_process().name})")
+
+    def _create_processor(self, entry: dict) -> "Processor":
+        processor_name = list(entry.keys())[0]
+        entry[processor_name]["metric_labels"] = self._metric_labels
+        processor = Factory.create(entry, self._logger)
```

Comment:
I know it's only indirectly related with this refactoring, but could you fix the type-hint of the factory? It currently has the return type `Connector` which conflicts with the type-hint of this method. So it should have a type like `Union[Connector, Processor]`.

Reply:
done